react/features/recording/googleApi.js

```diff
@@ -50,142 +66,159 @@ const googleApi = {
     },
 
     /**
-     * Checks whether the user is currently authenticated into google.
+     * Checks whether a user is currently authenticated with the Google Web
+     * Client configured in the Google API Client Library.
      *
      * @returns {boolean}
      */
     isSignedIn() {
-        const globalGoogleApi = this.getGoogleApiGlobal();
+        const googleApiClient = this._getGoogleApiClient();
 
         return Boolean(
-            globalGoogleApi
-            && globalGoogleApi.auth2
-            && globalGoogleApi.auth2.getAuthInstance
-            && globalGoogleApi.auth2.getAuthInstance().isSignedIn
-            && globalGoogleApi.auth2.getAuthInstance().isSignedIn.get()
+            googleApiClient
```

Comment:
here we don't load the googleApiClient if it's null but in the getter above we do...do we care about it in the `get` but don't want to load it (if it isn't already loaded) here?

Reply:
I think this goes with my todo item of refactoring the googleApi wrapper. I thought I had done an okay pass this morning but it's apparent now that I haven't. Do you think it'd be good to have it so everything in `googleApi` calls `get`? That would mean a check like `isSignedIn` would return a promise instead of a boolean, which is why I was hesitant to use the `get`. The big pro is consistency. (This is all based on the assumption that async loading of the google api client is a good plan.)